examples/InfiniteTalk/infinitetalk/wan/distributed/xdit_context_parallel.py

```diff
@@ -0,0 +1,630 @@
+    kwargs = dict(
+        e=e0,
+        seq_lens=seq_lens,
+        grid_sizes=grid_sizes,
+        freqs=self.freqs,
+        context=context,
+        context_lens=context_lens,
+        audio_embedding=audio_embedding,
+        ref_target_masks=token_ref_target_masks,
+        human_num=human_num,
+    )
+
+    if self.enable_teacache:
+        if self.cnt % 3 == 0:
+            if not should_calc_cond:
+                x += self.previous_residual_cond
+            else:
+                ori_x = x.clone()
+                for block in self.blocks:
+                    x = block(x, **kwargs)
+                self.previous_residual_cond = x - ori_x
+        elif self.cnt % 3 == 1:
+            if not should_calc_drop_text:
+                x += self.previous_residual_drop_text
+            else:
+                ori_x = x.clone()
+                for block in self.blocks:
+                    x = block(x, **kwargs)
+                self.previous_residual_drop_text = x - ori_x
+        else:
+            if not should_calc_uncond:
+                x += self.previous_residual_uncond
+            else:
+                ori_x = x.clone()
+                for block in self.blocks:
+                    x = block(x, **kwargs)
+                self.previous_residual_uncond = x - ori_x
+    else:
+        for block in self.blocks:
+            x = block(x, **kwargs)
+
+    # head
+    x = self.head(x, e)
+
+    # Context Parallel
+    x = get_sp_group().all_gather(x, dim=1)
+
+    # unpatchify
+    x = self.unpatchify(x, grid_sizes)
+    if self.enable_teacache:
+        self.cnt += 1
+        if self.cnt >= self.num_steps:
+            self.cnt = 0
+
+    return torch.stack(x).float()
+
+
+def usp_attn_forward_multitalk(self, x, seq_lens, grid_sizes, freqs, dtype=torch.bfloat16, ref_target_masks=None):
+    b, s, n, d = *x.shape[:2], self.num_heads, self.head_dim
+    half_dtypes = (torch.float16, torch.bfloat16)
+
+    def half(x):
+        return x if x.dtype in half_dtypes else x.to(dtype)
+
+    # query, key, value function
+    def qkv_fn(x):
+        q = self.norm_q(self.q(x)).view(b, s, n, d)
+        k = self.norm_k(self.k(x)).view(b, s, n, d)
+        v = self.v(x).view(b, s, n, d)
+        return q, k, v
+
+    q, k, v = qkv_fn(x)
+
+    # Use Gaudi-specific rope as complex type not supported on HPU.
+    q = rope_apply_gaudi(q, grid_sizes, freqs).to(q.device)
+    k = rope_apply_gaudi(k, grid_sizes, freqs).to(q.device)
+
+    # Context Parallel
+    k = get_sp_group().all_gather(k, dim=1)
+    v = get_sp_group().all_gather(v, dim=1)
+
+    x = self.fav3.forward(
+        half(q), half(k), half(v), cp_size=get_sequence_parallel_world_size(), layout_head_first=False
+    )
+    htcore.mark_step()
+
+    # output
+    x = x.flatten(2)
+    x = self.o(x)
+
+    x_ref_attn_map = None
+
+    return x, x_ref_attn_map
+
+
+def sp_crossattn_multi_forward(
+    self,
+    x: torch.Tensor,
+    encoder_hidden_states: torch.Tensor,  # 1, 21, 64, C
```

Comment:
Can you remove this kind of comment or use generic symbol?
The same one in Line 501/555